### ISSUE TITLE
Handle other languages package managers

### ISSUE THREAD

**Issue by drazik** [frontsideair/yarnhook#38]
Hi,

I was going to implement a git hook to do just the same thing that yarnhook does, but for Composer (PHP package manager). But I thought it would be a shame to have yarnhook for yarn, and something specific for Composer. So I'm wondering if you are opened to handling not node-related package managers? In which case I can open a PR to add composer.

Thanks!

**Comment by devinrhode2** [comment]
I know people using yarn+composer, and I'm sure they would be interested in
this! If @frontsideair opts out you could rename your fork to something
like "depsihook" where the i means install like with npm i. Or something
more conservative+verbose like "auto-install-deps-hook" or
"auto-install-packages-hook". If we want to get really fun signature name,
maybe "auto-depsi" or "depsi-auto" or "packagi" or "pkgier".

"auto" may conflate it with things like the web pack auto installer plugins

**Comment by drazik** [comment]
Yes, composer + npm/yarn must be a common use case. Mine is a Symfony project.

**Comment by frontsideair** [comment]
I have no objections to this. I personally don't use PHP but since composer hook will only work if there's a composer lockfile I think it is unobtrusive and see no reason to not allow this. So the PR is more than welcome.

In the future if the number of lockfiles proliferate and maintenance becomes a problem I may consider scaling back but for now it's all right to me!

**Comment by devinrhode2** [comment]
INSTALL ALL THE THINGS!

Yeah I don't think one more file check on some git operations will be
noticeable for anyone. The git hooks don't run often anyway. Even `ls`
appears to complete instantly, and then you have a list of all possible
lock files.

**Comment by drazik** [comment]
Nice. I'm going to open a PR very soon then :)
